clients/core/src/main/scala/mqperf/Receiver.scala

```diff
@@ -19,32 +20,63 @@ class Receiver(config: Config, mq: Mq, clock: Clock) extends StrictLogging {
   private val FinishWhenNoMessagesAfter = 60.seconds
 
   def run(): Future[Unit] = {
-    val mqReceiver = mq.createReceiver(config)
-    run(mqReceiver, None).flatMap(_ => mqReceiver.close())
+    val receiverFactory = mq.createReceiverFactory(config)
+    Future
+      .sequence {
+        (1 to config.receiversNumber)
+          .map(_ => {
+            receiverFactory.createReceiver()
+          }) // prepare n receivers (n = receiversNumbers)
+          .map(receiver => { // run iteration for each receiver on a separate future and close the connection afterwards
+            runIterations(receiver)
+              .flatMap(_ => receiver.close())
+          })
+      }
+      .flatMap { _ =>
```

Comment:
maybe we should be closing `.onCompelte` so that the close is run also when there's an error?

Reply:
I've used `andThen`. Is it fine?